primitives/src/lib.rs

```diff
@@ -149,3 +150,16 @@ pub trait EmergencyPriceFeeder<CurrencyId, PriceWithDecimal> {
 pub trait ExchangeRateProvider {
     fn get_exchange_rate() -> Rate;
 }
+
+/// Returns the actual quantity of tokens that a given integral amount represents.
+pub fn token_units<T>(amount: T) -> Result<T, DispatchError>
+where
+    T: CheckedAdd + CheckedMul + From<u8>,
+{
+    let mut rslt = amount;
+    for _ in 0..TOKEN_DECIMALS {
+        let rslt_opt = rslt.checked_mul(&T::from(10u8));
```

Comment:
rslt is for result? I prefer a full name :)

Reply:
Renamed